src/query/expression/src/schema.rs

```diff
@@ -215,46 +215,6 @@ impl DataSchema {
         Self::new_from(fields, self.meta().clone())
     }
 
-    /// project with inner columns by path.
```

Comment:
Is this code not used any more?

Reply:
Only `TableSchema` implement these methods.

Reply:
Ok, so we should modify `leaf_fields` as above.

Reply:
It seems we don't need to this for Array types. If we select array[i]:field, it will turn to call `get` method.